instrumentation/methods/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/methods/MethodInstrumentation.java

```diff
@@ -56,31 +59,46 @@ public ElementMatcher<TypeDescription> typeMatcher() {
   @Override
   public void transform(TypeTransformer transformer) {
     transformer.applyAdviceToMethod(
-        namedOneOf(methodNames.toArray(new String[0])).and(isMethod()),
+        namedOneOf(methodNames.keySet().toArray(new String[0])).and(isMethod()),
         mapping ->
-            mapping.bind(
-                MethodReturnType.class,
-                (instrumentedType, instrumentedMethod, assigner, argumentHandler, sort) ->
-                    Advice.OffsetMapping.Target.ForStackManipulation.of(
-                        instrumentedMethod.getReturnType().asErasure())),
+            mapping
+                .bind(
+                    MethodReturnType.class,
+                    (instrumentedType, instrumentedMethod, assigner, argumentHandler, sort) ->
+                        Advice.OffsetMapping.Target.ForStackManipulation.of(
+                            instrumentedMethod.getReturnType().asErasure()))
+                .bind(
+                    SpanKindOrdinal.class,
+                    (instrumentedType, instrumentedMethod, assigner, argumentHandler, sort) ->
+                        Advice.OffsetMapping.Target.ForStackManipulation.of(
+                            JavaConstant.Simple.ofLoaded(
```

Comment:
Looks like `new EnumerationDescription.ForLoadedEnumeration(methodNames.get(instrumentedMethod.getName()))` would let us directly use the enum constant without the need to mess with the ordinal